### ISSUE TITLE
Unable to compile binding project

### ISSUE THREAD

**Issue by greg-trak** [megsoftconsulting/Xamarin.Pushy.Android#1]
Getting the following errors when trying to compile the binding project:
Xamarin.Pushy.Android-master\src\Pushy.Android.BindingLibrary\Pushy.Android.BindingLibrary\obj\Debug\generated\src\ME.Pushy.Sdk.Services.PushySocketService.cs(25,24,25,36): error CS0534: 'PushySocketService.ConnectAsync' does not implement inherited abstract member 'AsyncTask.DoInBackground(params Object[])'

and

Xamarin.Pushy.Android-master\src\Pushy.Android.BindingLibrary\Pushy.Android.BindingLibrary\obj\Debug\generated\src\ME.Pushy.Sdk.Services.PushySocketService.cs(493,24,493,42): error CS0534: 'PushySocketService.SendKeepAliveAsync' does not implement inherited abstract member 'AsyncTask.DoInBackground(params Object[])'

Any idea how to resolve these issues?

**Comment by ajpinedam** [comment]
Just downloaded the code and I am facing the same issue when trying to do a clean build.

Was able to make it work modifying manually the generated file.

Double click on the errors you receive and search the code below and make those changes. It will compile fine unless you clean and rebuild. At least for the moment that could allow you create a package or a regular dll which doesn't requires rebuild for the moment. This is a workaround until I can see what's going on.

**class ConnectAsync
line: 97**

```
//change this
protected virtual unsafe global::Java.Lang.Integer DoInBackground (params global:: Java.Lang.Integer[] p0)

//to this
protected override unsafe global::Java.Lang.Object DoInBackground (params global:: Java.Lang.Object[] p0)

```

**class SendKeepAliveAsync
line: 565**

```
//change this
protected virtual unsafe global::Java.Lang.Integer DoInBackground (params global:: Java.Lang.Integer[] p0)

//to this
protected override unsafe global::Java.Lang.Object DoInBackground (params global:: Java.Lang.Object[] p0)
```

Doing the changes above I was able to create a package and tested and it works fine.

If you have problems let me know and I could publish the package I just created so you can use it.

**Comment by eladnava** [comment]
Hey @ajpinedam,
Thanks so much for your hard work on this project! 👍 

I am the Founder at Pushy and would like to personally thank you for the time and effort you put into making the Pushy Android SDK compatible with Xamarin, and for porting the demo app functionality.

I had a chance to investigate this issue further and found the recommended solution, according to a [Xamarin engineer](https://gist.github.com/JonDouglas/dda6d8ace7d071b0e8cb). Since the automatically-generated C# bindings are incorrect for these two internal Java `AsyncTask.doInBackground` declarations, we can simply ask Xamarin Studio to remove these bindings (avoid exposing them to the C# compiler) via the [`Transforms/Metadata.xml`](https://github.com/megsoftconsulting/Xamarin.Pushy.Android/blob/fde781d15ff3c0e978ed7c124a6b5df4c2b7ab4c/src/Pushy.Android.BindingLibrary/Pushy.Android.BindingLibrary/Transforms/Metadata.xml) file:

```
<?xml version="1.0" encoding="UTF-8"?>
<metadata>
   <remove-node path="/api/package[@name='me.pushy.sdk.services']/class[@name='PushySocketService.ConnectAsync']" />
   <remove-node path="/api/package[@name='me.pushy.sdk.services']/class[@name='PushySocketService.SendKeepAliveAsync']" />
</metadata>
```

This is absolutely fine since these classes are internal to the SDK and do not need to be exposed to the C# compiler.

We do intend to release an official Xamarin Binding Library for the Pushy Android SDK along with a sample project just as you have. Your research has helped us make this possible, so thank you! 💯

**Comment by ajpinedam** [comment]
@eladnava  

That's good news! Yeah Pushy is a great library/service and it will be greater now that you are making it available for others platforms.

If you need I'd be more than glad to give you a hand on this. 

Also, I have a well detailed documentation I created (never had the chance to put it here in the readme) based on the original documentation of Pushy for Android but more for Xamarin developers.

**Comment by greg-trak** [comment]
Thanks for your help on this guys.  We were able to get our POC for Pushy up and going.  Seems like a great service!

**Comment by eladnava** [comment]
@greg-trak My pleasure! Great to hear you are impressed with our service so far.

@ajpinedam Thank you so much for the offer!

We have actually just finished finalizing the [Xamarin.Android documentation](https://pushy.me/docs/additional-platforms/xamarin-android) and releasing an official [NuGet package](https://www.nuget.org/packages/Pushy.SDK.Xamarin.Android/).

We have also released a [demo app for Xamarin.Android](https://github.com/pushy-me/pushy-demo-xamarin-android).

This project has helped us a lot in the process of writing the docs and building the demo app, so thanks again! 👍
